sklearn/utils/tests/test_array_api.py

```diff
@@ -777,3 +778,24 @@ def test_logsumexp_like_scipy_logsumexp(array_namespace, device_, dtype_name, ax
         res_xp_2 = _logsumexp(array_xp_2, axis=axis)
         res_xp_2 = _convert_to_numpy(res_xp_2, xp)
         assert_allclose(res_np_2, res_xp_2, rtol=rtol)
+
+
+@pytest.mark.parametrize(
+    ("namespace", "device_", "expected_types"),
+    [
+        ("numpy", None, ("float64", "float32", "<class 'numpy.float16'>")),
```

Comment:
With numpy we get dtype(float64) and dtype(float32) from the array api function but we add float16 ourselves as xp.float16, which is why these string representations differ.